File: 10k app/frontend/lib/data/models/exercise.dart
class Exercise {
  final int id;
  final String name;
  final String exerciseType;
  final String muscleGroup;

  const Exercise({
    required this.id,
    required this.name,
    required this.exerciseType,
    required this.muscleGroup,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'] as int,
      name: json['name'] as String,
      exerciseType: json['exerciseType'] as String,
      muscleGroup: json['muscleGroup'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    'exerciseType': exerciseType,
    'muscleGroup': muscleGroup,
  };
}

File: 10k app/backend/src/main/java/com/gymflow/model/Exercise.java
package com.gymflow.model;

import com.gymflow.model.enums.ExerciseType;
import com.gymflow.model.enums.MuscleGroup;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "exercise")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Exercise {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ExerciseType exerciseType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MuscleGroup muscleGroup;
}
